indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -213,6 +218,13 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
           activeTasks.remove(task.getId());
         }
       }
+      activePendingTaskGroupToTaskIds.clear();
+      for (Task task : storedActiveTasks) {
+        if (activeTasks.contains(task.getId()) && task.getPendingSegmentGroup() != null) {
```

Comment:
Curious - can `task.getPendingSegmentGroup()` ever be null? Or is this more of a defensive check?

Reply:
It can be null for pending segments that were created before these changes. Same goes for parent_id